drivers/flash/soc_flash_nrf5.c

```diff
@@ -18,6 +18,7 @@
 #if defined(CONFIG_SOC_FLASH_NRF5_RADIO_SYNC)
 #include <misc/__assert.h>
 #include <bluetooth/hci.h>
+#include "controller/hal/nrf5/ticker.h"
```

Comment:
@cvinayak I leave the comment here, but it is rather generic: we introduce direct dependency, here, between two different drivers, over HAL. Is this inline with general conventions? Isn't it more normal to include the public ticker header file here?

Reply:
ticker is not public API. The convention followed here is inline with how applications using kernel include header files, example for ARM (below order among tons of other include):
- include/arch/arm/cortex_m/cmsis.h 
- include/kernel.h

Since ticker is private, header files are not self-contained, and dependent headers are included on requiring c implementation.